analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -189,10 +189,8 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
         }
 
         private fun parseScm(scm: Scm?): VcsInfo {
-            val connection = scm?.connection.orEmpty()
-            val tag = scm?.tag?.takeIf { it != "HEAD" }.orEmpty()
-
-            if (connection.isEmpty()) return VcsInfo.EMPTY
+            val connection = scm?.connection ?: return VcsInfo.EMPTY
+            val tag = scm.tag?.takeIf { it != "HEAD" }.orEmpty()
```

Comment:
You change functionality here, previously `VcsInfo.EMPTY` was also returned if `scm.connection` was empty, now only if it is null. I would suggest to use `val connection = scm?.connection?.takeIf { it.isnotEmpty() } ?: return VcsInfo.EMPTY`.
What I don't know is if Maven sets `scm.connection` to null or to empty if an empty `<connection/>` tag is provided in the POM, so this might not make a difference.

Reply:
Good points. More thoughts: Maybe prefer `takeUnless { it.isEmpty() }` to express that the expected default is to take the value. Or be a bit more explicit and use

            val connection = scm?.connection
            if (connection.isNullOrEmpty()) return VcsInfo.EMPTY